docs: 2024H1 roadmap and why VoDa supports Ibis (#8184)
ibis-project/ibis: docs/posts/roadmap-2024-H1/index.qmd

---
title: "Ibis project 2024 roadmap"
author: "Cody Peterson"
date: "2024-02-10"
image: commits.png
categories:
    - blog
    - roadmap
    - community
---

## Overview

Welcome to the first public roadmap for the Ibis project! If you aren't familiar
with the background of Ibis or who supports it nowadays, we recommend reading
[why Voltron Data supports Ibis](../why-voda-supports-ibis/index.qmd) before the
roadmap below.

> ## 2024 roadmap

Reviewer: Besides the roadmap, do we also have a set of principles for prioritization/design decisions that we should keep in the public? I feel it's important for us to differentiate from an open-core approach to an open-standard approach.

Reviewer 2: I don't think we have anything like this, but I'm also not sure we need it at the moment.

I also don't think positioning Ibis using the open core versus open standards terminology is the right approach and adds linguistic overhead and terminology that isn't particularly relevant for many users.

Author: I do agree this is something we should have (probably a concept article) but don't want to block on the roadmap for it -- it is relatively covered in the "why VoDa supports Ibis blog" that we don't intend to build an open-core SaaS around Ibis or anything

Reviewer: ok, sg. The reason why I ask is when I spoke to an integrating partner yesterday, I sensed concerns or, at the minimum, a lack of information to understand Ibis' OSS motivation (vs backing company's commercial motivation). I think this blog post will help already; it's just something we'll have to continue to think about and to strenghen trust with the community. Ultimately, what works best is probably just focusing on providing value and exponential growth :)

We have a [public roadmap as a GitHub
project!](https://github.com/orgs/ibis-project/projects/5)

![Ibis roadmap](roadmap.png)

We are early in our use of this GitHub project, so please pardon any
disorganization as we get it up and running efficiently. In general, we have:

- **Roadmap view**: consisting of meta-issues in their respective repositories
 for high-level objectives of the Ibis project
- **Triage view**: consisting of new issues across Ibis project repositories
 that need to be triaged
- **Backlog view**: consisting of issues that have been triaged (assigned a
 priority) and are on the backlog
- **TODO view**: consisting of issues that are in progress or ready to be worked
 on soon
- **Label-specific views**: consisting of issues for specific labels, like
 documentation or a large refactor

> Right now, [the team at Voltron Data](../why-voda-supports-ibis/index.qmd) sets

Reviewer: Would be nice to include a statement to call for action (encourage audience to participate), and provide links to instructions.

Author: added

the roadmap and priorities. Over time as more contributors and organizations
join the project, we expect this process to diversify and become more
community-driven.

### Overall themes

> Our top five themes for 2024 include:

Reviewer: Love these themes. We should include a concise sentence to convey the value to the audience for each. For example, why many Ibis backends is important? because user can **implement once and execute anywhere**

Author: added sentences...open to more suggestions, I don't love all of them but it's something

1. **Ibis backends**: Ibis is a Python frontend for many backends. To continue
scaling to more backends, we need to complete a major rework of library
internals and stabilize the API for backend authors. Related work in this area
will make it easier than ever to create new Ibis backends and maintain them.
This work will also include improving backend interfaces for operations like
table creation, insertion, and upsertion.

2. **Ibis for ML**: Increasingly, data projects are ML projects. Ibis can
uniquely help with feature engineering and other ML tasks connecting your data
where it lives to ML models. We will continue to improve Ibis for ML use cases.

3. **Ibis for streaming data**: Ibis has only been for batch data until very
recently. With the addition of the first streaming backends, we will continue to
improve Ibis for streaming data use cases and bridge the gap between batch and
streaming data.

4. **Ibis for geospatial**: Ibis has a rich set of geospatial expressions, but
most backends do not implement them. We will continue to improve Ibis for
geospatial use cases and bridge the gap between geospatial data and other data
types.

> 5. **Ibis community**: Ibis is an open source project and we want to make it as

Reviewer: Should we include a point to differentiate us from other open-core community? We need to build strong trust in the community over time, and it starts with messaging.

Reviewer 2: I don't think we need to split hairs on the minutia of open source versus open core versus open standards. Both open core and open standards are open source, but they only make sense in the context of a product and Ibis isn't a product. I think the distinction is useful, but it belongs at the level of a given product not here.

refactor soon!
:::

### Ibis for ML preprocessing

Data projects are increasingly ML projects. pandas and scikit-learn are the
default for Python users, but tend to lack scalability. Many projects look to
address this and Ibis does not intend on duplicating effort here. Instead, we
want to leverage what sets Ibis apart -- the ability to have a single Python API
that scales across many backends -- to feature engineering and other ML
preprocessing tasks ahead of model training.

Jim took this on over the last few months, building up the
[IbisML](https://github.com/ibis-project/ibisml) package to a usable (but still
toy) state. We will further invest in IbisML this year to get it a
production-ready state, bringing the power of Ibis to ML feature engineering.

We're [excited to welcome the (former) Claypot AI team to
VoDa](https://voltrondata.com/resources/voltron-data-acquires-claypot-ai) to
help drive this work forward! Expect a release announcement for IbisML soon
covering the majority of feature engineering operations and handoff to popular
ML training frameworks.

::: {.callout-note collapse="true" title="LLMs: the Ibis Birdbrain project"}

> I've been working on a new LLM integration for Ibis called `ibis-birdbrain`.

Reviewer 3: ```suggestion
Cody has been working on a new LLM integration for Ibis called `ibis-birdbrain`.
```

theme. We aim to keep the website simple and focused on documentation that helps
users, but want to deviate from the default themes in Quarto to make Ibis stand
out.

### Documentation

> "When you're ~~selling~~ distributing free and open source software, the
> documentation is the product." - old tech adage, origin unknown

A few months ago, we moved our documentation to [Quarto](https://quarto.org) and
revamped most of the website along the way. We will continue improving the
documentation with backend-specific getting started tutorials, how-to guides for
common tasks, improved API references, improving the website search
functionality, and more!

Improving the documentation is a great way to get involved with Ibis!

## Beyond Q1 2024

This writeup of our roadmap is heavily biased toward Q1 of 2024. Looking out,
our priorities remain much the same. After the big refactor is done, we will
continue improving our library internals, backend interface, and ensuring the
longevity of Ibis. We'll continue improving ML, streaming, and geospatial
support.

Expect an updated roadmap blog in the second half of the year for more details!

## Next steps

> It's never been a better time to get involved with Ibis. [Join us on Zulip and

Reviewer: Let's spell out what "involved with Ibis" means. It can either be contributing new backends (for xxx value), or integrating Ibis into your own framework (for yyy value), or using ibis as a pandas replacement for (zzz value), etc.

> It's never been a better time to get involved with Ibis. [Join us on Zulip and

Reviewer: also curious why we decide to use Zulip, somebody asked me today and i didn't know the answer.

Author: yep we had some internal discussion here -- Discord seemed outright wrong with how it handles threads/code snippets. Slack could have been an option (and we can still consider it), but would probably want a paid version to retain history. Zulip is a free and open source option that fits with Ibis will and has a lot of useful features

full-time? Why does Voltron Data employ five software engineers to work on Ibis
full-time? Great questions!

::: {.callout-note}
To understand Voltron Data -- or if you're generally interested in learning
about the composable data ecosystem -- check out the [The Composable
Codex by Voltron Data](https://voltrondata.com/codex).
:::

Voltron Data is a for-profit company building
[Theseus](https://voltrondata.com/theseus), an accelerator-native query engine
for the composable data ecosystem. On its own, Theseus is fairly bare bones -- it
needs storage below it, a frontend above it, and a bunch of other components to
connect things together. Voltron Data's founders and engineers are very
experienced with open-source software and pioneers of the composable data
ecosystem. Voltron Data supports Ibis because it is the Python frontend for
Theseus and can act as a standard Python dataframe API for **any** backend,
whether you're querying a CSV file on your laptop, running thousand-node Spark
jobs on a cluster in the cloud, or doing cutting-edge work on-premise with GPU
clusters. With Ibis, you can write your experimentation code for your laptop and

> scale up to Theseus (or any other backend) seamlessly.

Reviewer 2: I think the suggested replacement is great, but it's teetering on the edge of marketing. That has its place but I don't think the 2024 roadmap is the right place for this content. It's more suited for a blog post about why VoDa sponsors Ibis development.

Can we boil the content down here to one or two sentences?

> scale up to Theseus (or any other backend) seamlessly.

Reviewer 2: Perhaps we kill this entire section and write a blog post solely about why VoDa sponsors Ibis development.

> scale up to Theseus (or any other backend) seamlessly.

Reviewer 3: +1

@lostmygithubaccount what do you think? Do you want to pull this stuff into a separate blog post and I can help you flesh it out? We could link to it here.

> scale up to Theseus (or any other backend) seamlessly.

Author: let's go with that. I'll get a PR soon + adjust this to remove the first half of the content

The founders of Voltron Data experienced these pains with the pandas API
themselves in previous efforts, including cuDF. For Theseus and as an
open-source standard, we believe Ibis is the right approach.

Instead of using Snowpark Python for Snowflake, you can use Ibis on Snowflake.
Instead of using PySpark or pandas on Spark, you can use Ibis on Spark. Instead
of using the pandas API on BigQuery (built on top of Ibis), you can use Ibis on
BigQuery. Instead of using PyStarburst on Starburst Galaxy, you can use Ibis on
Starburst Galaxy. Instead of using the Polars Python on the Polars execution
engine, you can use Ibis on Polars. Instead of using DataFusion Python on
DataFusion execution engine, you can use Ibis on DataFusion. Instead of
executing SQL strings on DuckDB through the Python client, you can use Ibis on
DuckDB. And so on...

Ibis brings a Python dataframe interface to data platforms that only have SQL,
and brings a standard Python dataframe interface to data platforms that have
their own Python dataframe interface. It is the only portable Python dataframe
that can serve as a standard across the data ecosystem.
:::

> Voltron Data supports Ibis for the same reason it supports Apache Arrow.
> Open-source standards make it easier to build the composable data ecosystem and
> reduce friction for data teams looking to swap out components as their needs
> change. If you're already using Ibis as your frontend, Apache Arrow for data
> interchange, and Substrait for intermediary representation, you can swap in
> Theseus with ease!

Reviewer 3: ```suggestion
So Voltron Data supports Ibis because it can serve as a universal Python
dataframe API for **any** backend engine, whether you need to query a CSV file
on your laptop, run a big ETL job in the cloud with Snowflake or Starburst, or
process hundreds of terabytes in minutes on an NVIDIA GPU cluster with Theseus.
With Ibis, you have the choice of 20+ backends, and the code you write is the
same regardless of which backend you choose to use.
```

Reviewer 2: Our style guide says to avoid passive voice. Can we change `can serve` to `serves`?

Reviewer 4: nit pick: Can we also remove the "So" at the beginning of the sentence? I don't think it's needed here, and it clutters the sentence.

Reviewer 3: `can serve` is not passive voice.  Passive voice would be if it said `can be served by Ibis`

::: {.callout-important}
[Ibis is independently governed](https://github.com/ibis-project/governance) and

> not owned by Voltron Data. While currently four out of five members of the
> steering committee are employed by Voltron Data (the fifth being at Alphabet
> working on Google BigQuery), we expect this to change over time as more
> organizations join the project. **Ibis will never be solely available for VoDa
> products.**
> 
> Voltron Data also welcomes this dilution of power and influence! A healthy
> open-source project is one that is not controlled by a single entity. This is
> true of [Apache Arrow](https://arrow.apache.org) and other open-source projects
> that Voltron Data employees have been instrumental in building.

Reviewer 3: ```suggestion
not owned by Voltron Data. Currently, four out of five members of the
steering committee are employed by Voltron Data (the fifth being at Alphabet
working on Google BigQuery). We intend to work toward a more diverse
representation of companies and organizations as the project continues to grow.
```

Reviewer 2: We should say "are working toward" instead of "intend to work toward".

 team at VoDa recently focusing on [geospatial support in
 DuckDB](#geospatial-improvements)
- [**Phillip Cloud**](https://github.com/cpcloud): the tech lead for the Ibis
 team at VoDa

If you're interacting with us on GitHub or Zulip, you'll definitely run into at
least one of them! They make the Ibis project the delightful software it is
today and are always happy to help.

### Who else supports Ibis?

Anybody who contributes to Ibis is a supporter of Ibis! You can contribute by
[opening an issue](https://github.com/ibis-project/ibis/issues), [submitting a
pull request](https://github.com/ibis-project/ibis/pulls), [using Ibis in your
project](https://github.com/ibis-project/ibis/network/dependents), or [joining
the Zulip chat](https://ibis-project.zulipchat.com) to discuss problems or
ideas.

Notable organizations that support Ibis include:

- [**Claypot AI**](https://www.claypot.ai/): contributing the Apache Flink
 backend
- [**Exasol**](https://www.exasol.com/): contributing the Exasol backend
- [**Google's BigQuery
 DataFrames**](https://github.com/googleapis/python-bigquery-dataframes): a
 pandas API for BigQuery built on top of Ibis

> - [**RisingWave**](https://risingwave.com/): contributing the RisingWave backend

Reviewer 3: ```suggestion
- [**RisingWave**](https://risingwave.com/): contributing the RisingWave backend
- [**SingleStore**](https://github.com/singlestore-labs/ibis-singlestoredb): creating a SingleStore backend
```

---
title: "Ibis project 2024 roadmap"
author: "Cody"
date: "2024-02-05"
thumbnail: commits.png
categories:
    - blog
    - roadmap
    - community
---

## Overview

The Ibis project is an [independently governed open-source
community](https://github.com/ibis-project/governance) that builds the portable

> Python dataframe library. It is primarily backed by [Voltron
> Data](https://voltrondata.com) but has
> [contributors](https://github.com/ibis-project/ibis/graphs/contributors) across
> a range of data companies and institutions.

Reviewer 3: ```suggestion
Python dataframe library. Ibis has
[contributors](https://github.com/ibis-project/ibis/graphs/contributors)
across a range of data companies and institutions. Today the core Ibis
maintainers are employed by [Voltron Data](https://voltrondata.com).
```

This is the first public roadmap for the Ibis project. Thus, there's a bit of
backstory to explain how we got here and setup the context for where we're

> going. Feel free to [skip ahead to the roadmap](#2024-roadmap) if you're already

Reviewer 2: Clicking on `skip ahead to the roadmap` doesn't jump to the roadmap for me. Is that just a me thing?

Reviewer 2: Ah, the anchor link is just `#roadmap`, must be a quarto thing?

Reviewer 2: ![image](https://github.com/ibis-project/ibis/assets/417981/53b77005-65e0-4a34-bcf7-0d656dd1d56a)

Reviewer 2: ```suggestion
going. Feel free to [skip ahead to the roadmap](#roadmap) if you're already
```

Author: same...looking at the URL I guess I need to ignore the numbers, so just "#roadmap" here -- will change that, good catch

familiar with Ibis.

## Background

The Ibis project was started in 2015 by [Wes McKinney](https://wesmckinney.com),
the creator of pandas, as a pandas-like interface to Apache Impala. It received
improvements and support over the years, but really took off under the
stewardship of [Phillip Cloud](https://github.com/cpcloud) and the [current Ibis
team at Voltron Data](#who-are-the-core-contributors). It now supports 20+

> backends (including Apache Impala!) and is improving rapidly. It's never been a

Reviewer 2: ```suggestion
backends and is improving rapidly. It's never been a
```

better time to get involved with Ibis.

You can see the inflection point in the number of commits to the repository in
early 2022:

![Ibis commits over time](commits.png)

### Who am I?

My name is Cody and I'm employed by Voltron Data to work on Ibis full-time as a

> Technical Product Manager. I am an Ibis committer and have contributed Delta

Reviewer 4: I think the word "committer" sounds like you do sporadic work on. 
```suggestion
Technical Product Manager. I am an Ibis contributor and have worked on the Delta
```

Lake table input/output methods, helped moved the documentation over to
[Quarto](https://quarto.org), and created the [Zulip
chat](https://ibis-project.zulipchat.com) for the community.

My job is to help the Ibis community grow and thrive. I have a background in ML
(especially MLOps) and data products. Ibis solves many challenges I've seen in
the data space and I'm excited to help increase its adoption as a standard
Python frontend for dozens of data backends to reduce friction in the data
ecosystem.

### Why does Voltron Data (VoDa) support Ibis?

Why does Voltron Data employ a Technical Product Manager to work on Ibis

> full-time? Why does Voltron Data employ five software engineers to work on Ibis

Reviewer 4: Do we need the number "five" in here, given that some folks might be joining the team in the future?

Author: I'll add "currently" or something, but blogs are point in time. I was going to say "(and counting!)" or something but also didn't want to commit to that publicly

full-time? Great questions!

::: {.callout-note}
To understand Voltron Data -- or if you're generally interested in learning
about the composable data ecosystem -- check out the [The Composable
Codex by Voltron Data](https://voltrondata.com/codex).
:::

Voltron Data is a for-profit company building
[Theseus](https://voltrondata.com/theseus), an accelerator-native query engine

> for the composable data ecosystem. On its own, Theseus is fairly useless -- it

Reviewer 5: This seems like a good place for a metaphor, so it isn't just saying Theseus is useless (even with the caveat).  

beautiful engine sitting on the floor?
jet engine without a wing or a plane?

Author: metaphors are to me as getting hit in the head with a shovel is to me

> for the composable data ecosystem. On its own, Theseus is fairly useless -- it

Reviewer 2: ```suggestion
for the composable data ecosystem. On its own, Theseus is fairly bare bones -- it
```

needs storage below it, a frontend above it, and a bunch of other components to
connect things together. Voltron Data's founders and engineers are very
experienced with open-source software and pioneers of the composable data

> ecosystem. Thus, Voltron Data supports Ibis because it is the Python frontend

Reviewer 4: I don't think you need the "Thus" here.
```suggestion
ecosystem. Voltron Data supports Ibis because it is the Python frontend
```

> for Theseus and can act as a standard Python dataframe API for **any** backend,

Reviewer 2: ```suggestion
for Theseus and acts as a standard Python dataframe API for **any** backend,
```

The founders of Voltron Data experienced these pains with the pandas API
themselves in previous efforts, including cuDF. For Theseus and as an
open-source standard, we believe Ibis is the right approach.

Instead of using Snowpark Python for Snowflake, you can use Ibis on Snowflake.
Instead of using PySpark or pandas on Spark, you can use Ibis on Spark. Instead
of using the pandas API on BigQuery (built on top of Ibis), you can use Ibis on
BigQuery. Instead of using PyStarburst on Starburst Galaxy, you can use Ibis on
Starburst Galaxy. Instead of using the Polars Python on the Polars execution
engine, you can use Ibis on Polars. Instead of using DataFusion Python on
DataFusion execution engine, you can use Ibis on DataFusion. Instead of
executing SQL strings on DuckDB through the Python client, you can use Ibis on
DuckDB. And so on...

Ibis brings a Python dataframe interface to data platforms that only have SQL,
and brings a standard Python dataframe interface to data platforms that have
their own Python dataframe interface. It is the only portable Python dataframe
that can serve as a standard across the data ecosystem.
:::

> So, Voltron Data supports Ibis for the same reason it supports Apache Arrow.

Reviewer 5: ```suggestion
Voltron Data supports Ibis for the same reason it supports Apache Arrow.
```

> So, Voltron Data supports Ibis for the same reason it supports Apache Arrow.

Reviewer 4: No need for the "So," here
```suggestion
Voltron Data supports Ibis for the same reason it supports Apache Arrow.
```

- [**Jim Crist-Harif**](https://github.com/jcrist): the engineering manager for
 the Ibis team at VoDa
- [**Krisztián Szűcs**](https://github.com/kszucs): long-time Ibis contributor
 and primary author of the precursor to [the big refactor](#the-big-refactor)
- [**Naty Clementi**](https://github.com/ncclementi): newest member of the Ibis
 team at VoDa recently focusing on [geospatial support in
 DuckDB](#geospatial-improvements)
- [**Phillip Cloud**](https://github.com/cpcloud): the tech lead for the Ibis
 team at VoDa

If you're interacting with us on GitHub or Zulip, you'll definitely run into at
least one of them! They make the Ibis project the delightful software it is
today and are always happy to help.

### Who else supports Ibis?

Anybody who contributes to Ibis is a supporter of Ibis! You can contribute by
[opening an issue](https://github.com/ibis-project/ibis/issues), [submitting a
pull request](https://github.com/ibis-project/ibis/pulls), [using Ibis in your
project](https://github.com/ibis-project/ibis/network/dependents), or [joining
the Zulip chat](https://ibis-project.zulipchat.com) to discuss problems or
ideas.

Notable organizations that support Ibis include:

> - [**Claypot AI**](https://www.claypot.ai/): contributing the Apache Flink

Reviewer 4: Should be mentioned something about them being part of VoDa now? and maybe mention their names? 
I'm thinking that we will be putting efforts into streaming and ML and they'll most likely lead this efforts.

Author: it is mentioned a couple times below -- will re-look at this

Author: I'm going to leave it as a historical contributed the Flink backend pre-acquisition

improve the Ibis community and make it easier than ever to contribute to Ibis.

We believe these themes will help Ibis as a standard Python interface for many
backends and real-world data use cases.

### The big refactor

The biggest item in Q1 2024 and primary focus of the core Ibis team right now is
the big refactor -- dubbed "the epic split" -- continuing the great work
completed by Krisztián in [his PR splitting the relational
operations](https://github.com/ibis-project/ibis/pull/7752). You can read more
details in that PR, but the gist is that a new intermediary representation for
Ibis expressions is being has been created that drastically simplifies the
codebase.

With that refactor in place, each backend Ibis supports needs to be moved to the
new relational model. As a consequence, we are also swapping out SQLAlchemy for
[SQLGlot](https://github.com/tobymao/sqlglot). We are losing out on some of the
things SQLAlchemy did for us automatically, but overall this gives us a lot more
control over the SQL that is generated, reduces dependency overhead, and
simplifies the codebase further.

::: {.callout-note}

> By the time this is published, the refactor will likely be complete and on the

Reviewer 4: When are we planning on publishing this? Not sure if the "By the time this is published, the refactor..." statement will be true.

Author: yeah let me perhaps adjust

We would love to onboard new contributors to the project.

### New backends

Adding new backends is not a priority for the Ibis team at Voltron Data in Q1.
Instead, we are focusing on [the big refactor](#the-big-refactor) and other
internal library improvements to get Ibis to the point where adding new backends
is much easier and maintanable. That will take the form of stabilizing the new
intermediary representation, separating out **connection** from **compilation**
steps, and solidifying the API for backend authors. We will also introduce new
documentation and possibly testing frameworks to ease the burden of adding new
backends.

We are still happy to support new backends! Some have already been mentioned,
but being added in Q1 include:

- Apache Flink
- Exasol
- RisingWave

Adding a new backend is a great way to get involved with Ibis! If you're
interested, [join us on Zulip](https://ibis-project.zulipchat.com) and let us
know or [open an issue on
GitHub](https://github.com/ibis-project/ibis/issues/new/choose).

### Logo and website design

We will likely engage an external design firm to help us redesign the logo

> (initially created by Tim Swast, thanks Tim! it has served us well!) and website

Reviewer 2: ```suggestion
(initially created by Tim Swast, thanks Tim! It has served us well!) and website
```

theme. We aim to keep the website simple and focused on documentation that helps
users, but want to deviate from the default themes in Quarto to make Ibis stand
out.

### Documentation

> "When you're ~~selling~~ distributing free and open-source software, the
> documentation is the product." - old tech adage, origin unknown

A few months ago, we moved our documentation to [Quarto](https://quarto.org) and
revamped most of the website along the way. We will continue improving the
documentation with backend-specific getting started tutorials, how-to guides for
common tasks, improved API references, improving the website search
functionality, and more!

Improving the documentation is a great way to get involved with Ibis!

## Beyond Q1 2024

This writeup of our roadmap is heavily biased toward Q1 of 2024. Looking out,
our priorities remain much the same. After the big refactor is done, we will

> ensure common DDL support across backends (create table, insert, upsert) for

Reviewer 4: Expand DDL: I'd avoid using an acronym if you are only using it once. It can be confusing for people that are not familiarized with the lingo.

Reviewer 5: also, will we ensure common DDL support?

Reviewer 2: Let's hold off on publishing statements about broad DDL support. I don't think we've talked about specifics enough to make a decision about user-facing DDL uniformity.

ClickHouse's `CREATE TABLE` is an example that comes to mind. ClickHouse provides oodles of syntax that will simply never been supported in other backends. There _might_ be a core, but if we go the route of building an API we have to decide to do so collectively and ideally with a plan!

Reviewer 2: FWIW I think @lostmygithubaccount is alluding to the ability to `insert` PyArrow objects, which technically would fall under DML. Maybe it's okay to be specific and say something about `insert`?

Author: I'll reword

We will likely engage an external design firm to help us redesign the logo
(initially created by Tim Swast, thanks Tim! It has served us well!) and website

> theme. We aim to keep the website simple and focused on documentation that helps

Author: argh
